tests/std/tests/P0896R4_ranges_alg_uninitialized_value_construct/test.cpp

```diff
@@ -0,0 +1,143 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <algorithm>
+#include <cassert>
+#include <concepts>
+#include <memory>
+#include <ranges>
+#include <span>
+#include <utility>
+
+#include <range_algorithm_support.hpp>
+
+using namespace std;
+
+// Validate dangling story
+STATIC_ASSERT(same_as<decltype(ranges::uninitialized_value_construct(borrowed<false>{})), ranges::dangling>);
+
+enum class CanThrowAtDefaultConstruction : bool { no, yes };
+
+template <CanThrowAtDefaultConstruction CanThrow>
+struct int_wrapper {
+    inline static int constructions = 0;
+    inline static int destructions  = 0;
+
+    static void clear_counts() {
+        constructions = 0;
+        destructions  = 0;
+    }
+
+    static constexpr int magic_throwing_val = 4;
+    int val                                 = 10;
+
+    int_wrapper() noexcept(!static_cast<bool>(CanThrow)) {
+        ++constructions;

```

Comment:
We seem to be inconsistent about incrementing the construction counter before vs. after throwing. Could you audit these and make them consistent?

Reply:
After having removed the special casing It is consistent again